pkg/batch/subscriber/processor/push_sender.go

```diff
@@ -40,224 +38,132 @@ import (
-			return health.Unhealthy
-		}
-		return health.Healthy
+		logger:        logger,
 	}
 }
 
-func (s *sender) runWorker() error {
+func (p pushSender) Process(ctx context.Context, msgChan <-chan *puller.Message) error {
 	record := func(code codes.Code, startTime time.Time) {
-		handledCounter.WithLabelValues(code.String()).Inc()
-		handledHistogram.WithLabelValues(code.String()).Observe(time.Since(startTime).Seconds())
+		subscriberHandledCounter.WithLabelValues(subscriberPushSender, code.String()).Inc()
+		subscriberHandledHistogram.WithLabelValues(
+			subscriberPushSender,
+			code.String(),
+		).Observe(time.Since(startTime).Seconds())
 	}
 	for {
 		select {
-		case msg, ok := <-s.puller.MessageCh():
+		case msg, ok := <-msgChan:
 			if !ok {
 				return nil
 			}
-			receivedCounter.Inc()
+			subscriberReceivedCounter.WithLabelValues(subscriberPushSender).Inc()
 			startTime := time.Now()
 			if id := msg.Attributes["id"]; id == "" {
 				msg.Ack()
 				record(codes.MissingID, startTime)
 				continue
 			}
-			s.handle(msg)
+			p.handle(msg)
 			msg.Ack()
 			record(codes.OK, startTime)
-		case <-s.ctx.Done():
+		case <-ctx.Done():
 			return nil
 		}
 	}
 }
 
-func (s *sender) handle(msg *puller.Message) {
-	event, err := s.unmarshalMessage(msg)
+func (p pushSender) handle(msg *puller.Message) {
+	event, err := p.unmarshalMessage(msg)
 	if err != nil {
 		msg.Ack()
-		handledCounter.WithLabelValues(codes.BadMessage.String()).Inc()
+		subscriberHandledCounter.WithLabelValues(subscriberPushSender, codes.BadMessage.String()).Inc()
 		return
 	}
-	featureID, isTarget := s.extractFeatureID(event)
+	featureID, isTarget := p.extractFeatureID(event)
 	if !isTarget {
 		msg.Ack()
-		handledCounter.WithLabelValues(codes.OK.String()).Inc()
+		subscriberHandledCounter.WithLabelValues(subscriberPushSender, codes.OK.String()).Inc()
 		return
 	}
 	if featureID == "" {
 		msg.Ack()
-		handledCounter.WithLabelValues(codes.BadMessage.String()).Inc()
-		s.logger.Warn("Message contains an empty FeatureID", zap.Any("event", event))
+		subscriberHandledCounter.WithLabelValues(subscriberPushSender, codes.BadMessage.String()).Inc()
+		p.logger.Warn("Message contains an empty FeatureID", zap.Any("event", event))
 		return
 	}
-	if err := s.send(featureID, event.EnvironmentNamespace); err != nil {
+	if err := p.send(featureID, event.EnvironmentNamespace); err != nil {
 		msg.Ack()
-		handledCounter.WithLabelValues(codes.NonRepeatableError.String()).Inc()
+		subscriberHandledCounter.WithLabelValues(subscriberPushSender, codes.NonRepeatableError.String()).Inc()
 		return
 	}
 	msg.Ack()
-	handledCounter.WithLabelValues(codes.OK.String()).Inc()
+	subscriberHandledCounter.WithLabelValues(subscriberPushSender, codes.OK.String()).Inc()
 }
 
-func (s *sender) send(featureID, environmentNamespace string) error {
+func (p pushSender) send(featureID, environmentNamespace string) error {
 	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Could you please increase the timeout to 30 seconds?

Reply:
Sure!